### ISSUE TITLE
[Bug] How to use `write` when using `win32` file handles

### ISSUE THREAD

**Issue by Smurf-IV** [cppfastio/fast_io#777]
Visual Studio 17.10
x64 stl/std project (No MFC)
The following code will not compile:
``` c++
            fast_io::obuf_file outfile_str( stressFileName, fast_io::open_mode::out | fast_io::open_mode::text | fast_io::open_mode::trunc );
            fast_io::write(outfile_str,utf8bom.cbegin(),utf8bom.cend()); 
```

States:
![image](https://github.com/user-attachments/assets/55a638cf-af6d-4527-9668-e52eff51c43f)

**Comment by trcrsired** [comment]
next branch?

**Comment by Smurf-IV** [comment]
I can comment on this repo no more, as I have trouble using this with linked projects that are also /clr builds.
Thankyou.

**Comment by trcrsired** [comment]
> I can comment on this repo no more, as I have trouble using this with linked projects that are also /clr builds. Thankyou.

what? clr builds?

**Comment by Smurf-IV** [comment]
Nope, it has been "Improved" to work correctly with the latest CRT and C++20:
![image](https://github.com/user-attachments/assets/65cc6756-b48c-419d-b402-e58cf8f5ea1a)

**Comment by trcrsired** [comment]
> Nope, it has been "Improved" to work correctly with the latest CRT and C++20: ![image](https://private-user-images.githubusercontent.com/2418812/351990146-65cc6756-b48c-419d-b402-e58cf8f5ea1a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5MjM1ODYsIm5iZiI6MTcyMTkyMzI4NiwicGF0aCI6Ii8yNDE4ODEyLzM1MTk5MDE0Ni02NWNjNjc1Ni1iNDhjLTQxOWQtYjQwMi1lNThjZjhmNWVhMWEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjVUMTYwMTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGEyNzE5Mjc3ZDM0Yjk2NWQ3ODRjZDEwMTM1MWMwZTMzYTcwZWZlMTQwNjRjOTBjYzY1MjZjNWQ3ZmVmY2FhNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.UIyDAOfkwpa0BX7EGorBeyzt5eElclzhsoNjT2poITI)

what about next branch with ::fast_io::operations::write_all_bytes?
